src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -77,23 +92,56 @@ const useFetchChannelHook = () => {
 
     try {
       const members = getChannelMembers(channel);
-      await Promise.all(
-        (members || []).map(async (member) => {
-          const userMember = UserSchema.fromMemberWebsocketObject(member, channel?.id);
-          await userMember.saveOrUpdateIfExists(localDb);
-        })
-      );
-
-      await Promise.all(
-        (channel?.messages || []).map(async (message) => {
-          if (message?.type === 'deleted') return;
-          if (message?.type === 'system') {
-            message = getFirstMessage([message]);
+
+      (members || []).map((member) => {
+        const userMember = UserSchema.fromMemberWebsocketObject(member, channel?.id);
+        queue.addJob({
+          label: `saveUserMember-${member?.user?.id}`,
+          task: () => {
+            return new Promise((resolve) => {
+              UserSchema.isUserExists(localDb, member?.user?.id, channel?.id).then(
+                (isUserExists) => {
+                  if (!member?.user?.username) return resolve(true);
+                  if (!isUserExists) {
+                    userMember.save(localDb).then(() => {
+                      return resolve(true);
+                    });
+                  } else {
+                    return resolve(true);
+                  }
+                }
+              );
+            });
           }
-          const chat = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
-          await chat.save(localDb);
-        })
-      );
+        });
+
+        return null;
+      });
+
+      (channel?.messages || []).map((message) => {
+        queue.addJob({
+          label: `saveChat-${message?.id}`,
+          task: () => {
+            return new Promise((resolve) => {
+              if (message?.type === 'deleted') {
+                resolve(true);
+                return;
+              }
+              if (message?.type === 'system') {
+                message = getFirstMessage([message]);
+              }
+              const chat = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
+              chat.saveIfNotExist(localDb).then(() => {
+                resolve(true);
+
+                refreshWithId('chat', channel?.id);
+              });
+            });
+          }
+        });
+
+        return null;
+      });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are significant and improve the code quality. The use of a queue for saving user members and chat messages asynchronously is a good practice as it can help prevent blocking of the main thread. However, similar to the previous comment, error handling within the job task could be improved. Consider adding error handling inside the task function.



<!-- This is an auto-generated comment by OSS CodeRabbit -->